src/UI/Icon.elm

```diff
@@ -0,0 +1,79 @@
+module UI.Icon exposing
+    ( Icon
+    , close
+    , toEl
+    , todo
+    , toggle
+    )
+
+import Element exposing (..)
+import Html
+import Html.Attributes as HtmlAttr
+import UI.Attributes exposing (title)
+import UI.RenderConfig exposing (RenderConfig)
+
+
+todo : String -> Element msg
+todo _ =
+    -- TODO: Remove
+    fasIcon "--------" ""
+
+
+type alias Properties =
+    { hint : String
+    , glyph : IconGlyph
+    }
+
+
+type Icon
+    = Icon Properties
+
+
+type IconGlyph
+    = Toggle
+    | Close
+
+
+toggle : String -> Icon
+toggle hint =
+    Icon (Properties hint Toggle)
+
+
+close : String -> Icon
+close hint =
+    Icon (Properties hint Close)
+
+
+toEl : RenderConfig -> Icon -> Element msg
+toEl _ (Icon { hint, glyph }) =
+    case glyph of
+        Toggle ->
+            fasIcon "map" hint
```

Comment:
should we call this Map instead of  `Toggle`?

Reply:
It's just a placeholder for a proper "toggling" icon.
I also believe they should be called by their function instead of what they look like. This way it's the developer's fault if it chooses to have an icon representing a wrong function.

Reply:
I think in some cases the same icon could be used for different functions, as will be used across different projects